crates/api/src/inmem.rs

```diff
@@ -359,27 +383,109 @@ impl EmbeddedChronicleTp {
             context: context.clone(),
         };
 
-        Self {
+        let listen_port = portpicker::pick_unused_port().expect("No ports free");
+        let listen_addr = SocketAddr::new(Ipv4Addr::new(127, 0, 0, 1).into(), listen_port);
+        let connect_addr = SocketAddr::new(Ipv4Addr::new(127, 0, 0, 1).into(), listen_port);
+
+        let behavior_clone = behavior;
+        thread::spawn(move || {
+            let rt = runtime::Builder::new_current_thread()
+                .enable_io()
+                .enable_time()
+                .build()
+                .unwrap();
+            let mut rx = UnboundedReceiverStream::new(rx);
+            let local = tokio::task::LocalSet::new();
+
+            let task = local.run_until(async move {
+                tokio::task::spawn_local(async move {
+                    let (mut router_tx, mut router_rx) = router(&Context::new())
+                        .bind(&format!("tcp://{}", listen_addr))
+                        .unwrap()
+                        .split();
+
+                    debug!(listen_addr = ?listen_addr, "Embedded TP listening");
+                    loop {
+                        select! {
+                            message = router_rx.next().fuse() => {
+                              if message.is_none() {
+                                break;
+                              }
+
+                              let mut multipart = message.unwrap().unwrap();
+
+                              debug!(request = ?multipart);
+                              let _request = multipart.pop_front().unwrap().to_vec();
```

Comment:
It's odd that we pop two and ignore the first. Perhaps some adjective in the naming might help illustrate what's going on?

Reply:
Blank message frames are meant to be used in mulipart by routers according to zmq

Reply:
Ah, maybe call it `_blank_request` or somesuch?